model/map/src/main/java/org/keycloak/models/map/storage/MapModelCriteriaBuilder.java

```diff
@@ -91,10 +91,11 @@ private MapModelCriteriaBuilder(Map<SearchableModelField<M>, UpdatePredicatesFun
         }
         Predicate<? super K> resIndexFilter = b.getKeyFilter() == ALWAYS_TRUE ? ALWAYS_TRUE : b.getKeyFilter().negate();
         Predicate<? super V> resEntityFilter = b.getEntityFilter() == ALWAYS_TRUE ? ALWAYS_TRUE : b.getEntityFilter().negate();
+
         return new MapModelCriteriaBuilder<>(
           fieldPredicates,
-          v -> keyFilter.test(v) && ! resIndexFilter.test(v),
-          v -> entityFilter.test(v) && ! resEntityFilter.test(v)
+          v -> keyFilter.test(v) && resIndexFilter.test(v),
+          v -> entityFilter.test(v) && resEntityFilter.test(v)
```

Comment:
This change is a bugfix? Could you please explain it for me?

Reply:
Yes this is a bug fix. Previously `not` method didn't work because `resIndexFilter` and `resEntityFilter` were double negated using `negate()`[1] and using `!`. We needed to remove one of them.

[1](https://github.com/keycloak/keycloak/pull/7878/files#diff-334de1c064180d49ec21d4d167874149862721e3b4994d956f334ae653a226d6R92-R93)